### PR TITLE
refactor(config)!: use `char` for single-letter types

### PR DESCRIPTION
**Pull request by KSXGitHub**
## Summary

The `single_letter_*` rules' `extra_allowed_idents` / `ignore_allowed_idents` knobs were typed as `Vec<String>`, but `is_single_ascii_letter` gates the match, so any multi-character or non-letter entry silently never fired.

## Changes

- `src/rules/single_letter_{let_binding,function_param,closure_param}.rs`:
  - retype `extra_allowed_idents` / `ignore_allowed_idents` as `Vec<char>`; defaults from `&[&str]` to `&[char]`.
  - field docstrings now state the `[A-Za-z]` constraint and the config-parse-time failure mode so the generated `rules/*.md` is accurate.
- `src/common.rs`:
  - add `deserialize_ascii_letters` (rejects any entry that isn't a single ASCII letter at TOML-parse time, so `["xy"]` or `["1"]` produces a config-parse error instead of a silent no-op) and `resolve_symbol_set_from_chars` (intern a `Vec<char>` to `BTreeSet<Symbol>`); the three rules wire their config through both.
  - add a `#[cfg(test)] mod tests` block pinning the deserializer's contract: empty list and ASCII letters succeed, a multi-codepoint string fails at serde-toml's `char` parser, ASCII digits and non-ASCII letters surface our custom error message.
- `tests/single_letter_{let_binding,function_param,closure_param}.rs`: mirror the new `Vec<char>` shape.
- `tools/gen-docs/src/extract/ty.rs`: split `char` out of the multi-character `string` family; it now renders as `single-character string` (honest for every current `Vec<char>` field, since serde-toml only accepts one-character TOML strings for `char`). A tighter `single-letter string` label is reserved for the planned `AsciiLetter(char)` newtype tracked in #120.
- `planned-rules/single-letter-{const-item,static-item,const-generic}.md`: documentation update — restate the planned `allowed_idents` knob as `[single-character string]` / `Vec<char>` so future implementations match.
- `rules/*.md`: regenerated by `just gen-rules-md`. The new label also surfaces on `unicode_ellipsis_in_{comments,panic_messages}.also_flag`, which already used `Vec<char>`.

## Breaking change

`dylint.toml` configurations that wrote multi-character strings in `extra_allowed_idents` / `ignore_allowed_idents` will now fail to deserialise instead of being silently ignored.

### CODE REVIEW

File: tests/single_letter_let_binding.rs [hunk 1]
```diff
@@ -24,9 +24,9 @@ static SERIAL: Mutex<()> = Mutex::new(());
 #[derive(Default, serde::Serialize)]
 struct RuleConfig {
     #[serde(skip_serializing_if = "Option::is_none")]
-    extra_allowed_idents: Option<Vec<String>>,
+    extra_allowed_idents: Option<Vec<char>>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    ignore_allowed_idents: Option<Vec<String>>,
+    ignore_allowed_idents: Option<Vec<char>>,
```
[the remark this answers is not in the document]
Comment (KSXGitHub) [reply tests/single_letter_let_binding.rs:29]: The deserializer's rejection contract is already pinned by the `#[cfg(test)] mod tests` block added to `src/common.rs` in `dfdf657`. Five tests exercise the same `deserialize_ascii_letters` function the rule's `Config` runs through:

- `multi_character_string_is_rejected_at_parse_time` — covers `["xy"]` (serde-toml's `Vec<char>::deserialize` rejects it).
- `ascii_digit_is_rejected_with_our_message` — covers `["1"]` (our validator rejects with the custom message).
- `non_ascii_letter_is_rejected_with_our_message` — covers `["é"]`.
- `empty_list_is_accepted` and `ascii_letters_are_accepted` for the happy paths.

The rule's `Config` deserialisation reaches the validator via `#[serde(deserialize_with = "deserialize_ascii_letters")]`, so the unit-test coverage is the same path. An integration `#[should_panic]` test would couple to `dylint_linting::config_or_default`'s specific error-surfacing shape (Result vs. panic vs. fatal log) without strengthening the contract, so I'm leaving the unit tests as the authoritative pin.

Docstrings on the three rules' `extra_allowed_idents` / `ignore_allowed_idents` fields have been updated in `d32ec45` to document the `[A-Za-z]` constraint and the failure mode, which addresses the sibling threads on the rule source files.

---
_Generated by [Claude Code](https://claude.ai/code)_
